### PR TITLE
#36: Adds a new configuration option to prevent the deletion of temporary directories

### PR DESCRIPTION
**Pull request by paulroemer**
A first try and base for discussion.

### CODE REVIEW

**Comment by paulroemer** [comment]
Hi Michael,

here my first suggestion. For now I just added some simple stupid tests that check if setting the parameter works.

What is discussable:
1) The name of the parameter
2) Am i allowed to refactor the shutdown handler to make testing possible and to add a mocking framework you like to test it?

Cheers,
  Paul

**Comment by vorburger** [comment]
@paulroemer thanks for taking this on! (And sorry for delayed reply.) This would work for me...

Re. 1. naming, how about deletesTemporaryFilesOnShutdown instead of deletesTemporaryDataOnShutdown? Or even deletesTemporaryBaseAndDataOnShutdown? Just because deletesTemporaryDataOnShutdown sounds a bit like its affecting the data dir only, when in fact it's for both (which is fine with me). Change it everywhere, if OK with you.

Re. 2. "allowed to refactor the shutdown handler" - of course! As long as everything still works, you can contribute whatever you think can be improved - most welcome. (Keep public API comparability in mind, for the main functionality; for the shutdown handler it shouldn't matter.)

Re. mocking framework, if you must use one, my preference is Mockito - BUT, personally, and I do realise this is not commonly accepted... ;-) I'm, in general, not a huge fan of funky mocking frameworks.. I've seen too many tests of people mocking too much. I just write simple unit tests where I often don't find the need for mocking frameworks, and do like to write integration tests, e.g. db.stop() and assert dir still there IFF deletesTemporaryFilesOnShutdown false...

... I see you can't easily do that, but ... in looking over my code again, I'm not sure anymore if at the time I had had a particular reason for doing this FileUtils.deleteQuietly() only in the ShutdownHook - if I did this again, I would seem more natural to me now to just always do this in the db.stop() instead? What do you think? If you like this, feel free to put in a change to move that - perhaps best in a separate PR first?

**Comment by vorburger** [comment]
@paulroemer if you have a chance to finish that up, I'd love to merge it! No stress, but FYI I'm probably going to push a new release "soon" with what's coming from https://github.com/vorburger/MariaDB4j/pull/71.

File: mariaDB4j-core/src/main/java/ch/vorburger/mariadb4j/DBConfigurationBuilder.java [hunk 1]
```diff
@@ -113,6 +114,22 @@ public DBConfigurationBuilder setPort(int port) {
         return this;
     }
 
+    public boolean deletesTemporaryDataOnShutdown() {
```
Comment (vorburger) [inline mariaDB4j-core/src/main/java/ch/vorburger/mariadb4j/DBConfigurationBuilder.java]: is...

**Comment by vorburger** [comment]
@paulroemer never got to this, so I'll see if I can finish this up, based on the proposals made above.

**Comment by vorburger** [comment]
Closing, as I've finished this off in https://github.com/vorburger/MariaDB4j/pull/84. Will do an IT separately in next.

@paulroemer I only realized looking over this now that you were @ vaadin.com ... :) I've used Vaadin in the past - good stuff! Cool to see you guys use my MariaDB4j presumably in one of your customer projects? Feel free to raise a PR to add you / Vaadin to the [USERS](https://github.com/vorburger/MariaDB4j/blob/master/USERS.md) and/or [CONTRIBUTORS](https://github.com/vorburger/MariaDB4j/blob/master/CONTRIBUTORS.md) files.

[Standard Footer] PS: Shameless plug, consider https://www.patreon.com/vorburger to support MariaDB4j.

**Pull request closed without merging**